js/foundation.offcanvas.js

```diff
@@ -87,8 +89,14 @@ class OffCanvas {
       'keydown.zf.offcanvas': this._handleKeyboard.bind(this)
     });
 
-    if (this.options.closeOnClick && this.$exiter.length) {
-      this.$exiter.on({'click.zf.offcanvas': this.close.bind(this)});
+    // If we have an overlay and close on click, let it close the off canvas menu.
+    if (this.options.closeOnClick && this.options.contentOverlay && this.$overlay.length) {
```

Comment:
I'm happy to adjust to whatever would prefer - whats your preference? Change everything to direct true/false? Its like this as thats what the existing off canvas used, so I went with that too.